pol-core/ecompile/ECompileMain.cpp

```diff
@@ -713,6 +820,250 @@ void parallel_process( const std::set<std::string>& files )
+             owners_itr != dependency_owners.end() )
+        {
+          // Erase filepath from set of owners
+          owners_itr->second.erase( filepath );
+
+          // If owners set is empty, this dependency is no longer used by
+          // anything.
+          if ( owners_itr->second.empty() )
+          {
+            //  Erase from map of dependency -> owners.
+            dependency_owners.erase( owners_itr );
+            listener.remove_file( depnamepath );
+            VERBOSE_PRINTLN( "Remove watch file {}", depnamepath );
+          }
+        }
+      }
+      owner_dependencies.erase( itr );
+    }
+  };
+
+  auto add_dir = [&]( const std::string& elem )
+  {
+    fs::path dir( elem );
+    if ( fs::exists( dir ) )
+    {
+      auto dirpath = fs::canonical( dir );
+      if ( listener.add_watch_dir( dirpath ) )
+      {
+        VERBOSE_PRINTLN( "Add watch dir {}", dirpath );
+      }
+    }
+  };
+
+  for ( const auto& dep_owners : dependency_owners )
+  {
+    listener.add_file( dep_owners.first );
+  }
+
+  for ( const auto& directory : compiled_dirs )
+  {
+    listener.add_dir( directory );
+  }
+
+  for ( const auto& elem : compilercfg.PackageRoot )
+  {
+    add_dir( elem );
+  }
+
+  add_dir( compilercfg.ModuleDirectory );
+  add_dir( compilercfg.IncludeDirectory );
+  add_dir( compilercfg.PolScriptRoot );
+
+  fileWatcher.watch();
+
+  while ( !Clib::exit_signalled )
+  {
+    std::this_thread::sleep_for( std::chrono::seconds( 1 ) );
+    listener.take_messages( watch_messages );
+    if ( !watch_messages.empty() )
+    {
+      for ( auto const& message : watch_messages )
+      {
+        const auto& filepath = message.filepath;
+        const auto& old_filepath = message.old_filepath;
+        VERBOSE_PRINTLN( "Event: filename={}, old_filename={}", filepath, old_filepath );
+
+        // created or modified
+        if ( old_filepath.empty() )
+        {
+          handle_compile_file( filepath );
+        }
+        // deleted
+        else if ( filepath.empty() )
+        {
+          handle_delete_file( old_filepath );
+        }
+        // moved
+        else
+        {
+          handle_delete_file( old_filepath );
+          handle_compile_file( filepath );
+        }
+      }
+      watch_messages.clear();
+
+      if ( !to_compile.empty() )
+      {
+        summary.CompiledScripts = 0;
+        summary.UpToDateScripts = 0;
+        summary.ScriptsWithCompileErrors = 0;
+        em_parse_tree_cache.clear();
+        inc_parse_tree_cache.clear();
+
+        VERBOSE_PRINTLN( "To compile: {}", to_compile );
+
+        Tools::Timer<> timer;
+        for ( const auto& filepath : to_compile )
+        {
+          std::set<fs::path> removed_dependencies;
+          std::set<fs::path> new_dependencies;
+          try
+          {
+            process_file_wrapper( filepath.generic_string().c_str(), &removed_dependencies,
+                                  &new_dependencies );
+          }
```

Comment:
Call here add_dependencies_info

Reply:
`add_dependency_info` is inside `process_file_wrapper `, so that when a file is compiled, dependency info is updated. Are you suggesting to move it?